google/resource_storage_bucket_iampolicy.go

```diff
@@ -0,0 +1,122 @@
+package google
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+
+	storagev1 "google.golang.org/api/storage/v1"
+)
+
+func resourceStorageBucketIAMPolicy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceStorageBucketIAMPolicyCreate,
+		Read:   resourceStorageBucketIAMPolicyRead,
+		Update: resourceStorageBucketIAMPolicyUpdate,
+		Delete: resourceStorageBucketIAMPolicyDelete,
+
+		Schema: map[string]*schema.Schema{
+			"bucket": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"policy_data": {
+				Type:             schema.TypeString,
+				Required:         true,
+				DiffSuppressFunc: jsonPolicyDiffSuppress,
+				ValidateFunc:     validateV2IamPolicy,
+			},
+			"etag": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func setBucketIamPolicy(d *schema.ResourceData, config *Config) error {
+	bucket := d.Get("bucket").(string)
+	policy, err := unmarshalStorageIamPolicy(d.Get("policy_data").(string))
+	if err != nil {
+		return fmt.Errorf("'policy_data' is not valid for %s: %s", bucket, err)
+	}
+
+	_, err = config.clientStorage.Buckets.SetIamPolicy(bucket, policy).Do()
+	return err
+}
+
+func resourceStorageBucketIAMPolicyCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	if err := setBucketIamPolicy(d, config); err != nil {
+		return err
+	}
+
+	d.SetId(d.Get("bucket").(string))
```

Comment:
If we set this to a combination of project and bucket, we won't need to change IDs to support import.

Reply:
I don't understand exactly what you want. Actualy i do not have project var.
As seen in resource_storage_bucket_acl, ican set bucket+ "-iam_policy"?

Reply:
So what I'm suggesting here and why:

* When we make resources importable, we need to be able to identify the resources based on their ID. This means the ID needs to not only be unique, but contain all the information necessary to retrieve the resource.
* For this resource, it looks like we need both the project and the bucket ID to be able to retrieve the bucket.
* So ideally, our ID would be something like `{project}/{bucket}`, which we could then parse into the info we need to be able to retrieve the resource given just its ID.

Reply:
On google Policy, i have etag parameter that is an entity ID (return by API). but this it does not contain project.
I don't see how to make it.